src/data/tx-chunks-data-source.ts

```diff
@@ -123,6 +127,21 @@ export class TxChunksDataSource implements ContiguousDataSource {
         });
       });
 
+      if (region) {
+        const byteRangeStream = new ByteRangeTransform(
+          region.offset,
+          region.size,
+        );
+        return {
+          stream: stream.pipe(byteRangeStream),
+          size: region.size,
```

Comment:
@dtfiedler @djwhitt I'm not sure if I can do this. Returning just a small part will not break indexing?